File: ruby/lib/bizx/chess_strategy.rb
module BizX
  class ChessStrategy
    attr_reader :turn, :resources
    def initialize
      @turn = 0
      @resources = { 'gold' => 100, 'food' => 100, 'science' => 0 }
    end
    def tick(resource, amount)
      @resources[resource] = (@resources[resource] || 0) + amount
      @turn += 1
    end
    def uci(fen, command = 'go movetime 100')
      { protocol: 'UCI', fen: fen, command: command, adapter: 'external-engine' }
    end
  end
end

File: python/bizx/chess_strategy.py
"""Native Python chess/strategy facade for BizX."""
from dataclasses import dataclass, field
from hashlib import sha256
from typing import Dict, List

@dataclass
class StrategyState:
    turn: int = 0
    resources: Dict[str, int] = field(default_factory=lambda: {"gold": 100, "food": 100, "science": 0})
    territory: List[str] = field(default_factory=list)

class ChessStrategy:
    def __init__(self):
        self.state = StrategyState()

    def strategy_tick(self, resource: str, amount: int) -> dict:
        self.state.resources[resource] = self.state.resources.get(resource, 0) + amount
        self.state.turn += 1
        return self.snapshot()

    def snapshot(self) -> dict:
        payload = f"{self.state.turn}|{sorted(self.state.resources.items())}|{self.state.territory}".encode()
        return {"turn": self.state.turn, "resources": dict(self.state.resources), "checksum": sha256(payload).hexdigest()}

    def chess_engine_command(self, fen: str, command: str = "go movetime 100") -> dict:
        return {"protocol": "UCI", "position": fen, "command": command, "adapter": "external-engine"}
